### DISCUSSION TITLE
OTA Upgrade with wrong image size

### DISCUSSION THREAD

**Discussion by pavkamlc** [arendst/Tasmota#16313] [Q&A] [answered]
Hi everyone,
 I have sonoff-zbbridge tasmotized by tuya-convert and this works (11.0.0). Problem is, device cant long time update newer tasmota.
 I found problem is in wrong image before for 4M version, but this device is only 2M

STATUS4 = {"StatusMEM":{"ProgramSize":673,"Free":1116,"Heap":26,"ProgramFlashSize":4096,"FlashSize":2048,"FlashChipId":"1540A1","FlashFrequency":40,"FlashMode":3,"Features":["00000809","0F1007C6","04400001","00000003","00000000","00000000","00020080","00200000","00000000"],"Drivers":"1,2,4,7,9,10,12,20,23,38,41,50","Sensors":"1"}}

 When I upload (12.1.0), I get:
 UPGRADE = {"Upgrade":"Failed Update error: ERROR[8]: Flash config wrong real: 2097152 IDE: 4194304"}

Is there way to flash right image without soldering? I found some discuss about Partition_Wizard.tapp, but it not installs for me.

Regards, ...

**Comment by sfromis** [comment]
Which specific binary file are you trying to upgrade to? The prefilled one on the upgrade page is likely incorrect.

**Comment by pavkamlc** [comment]
I think, correct binary as first step before tasmota-zbbridge.bin is this:
http://ota.tasmota.com/tasmota/tasmota-minimal.bin.gz

**Reply by sfromis** [reply]
Normally, you should not specify the -minimal binary manually, as Tasmota will automatically use it when relevant, when you upgrade to the URL for the upgrade binary (download first is not recommended). Thus the "correct" URL is `http://ota.tasmota.com/tasmota/tasmota-zbbridge.bin.gz`, as long as the version you come from is not too old. Anyway, I don't think that this solves your issue.....

(The Partition Wizard is only relevant for ESP32 devices like the newer Sonoff "pro" Zigbee bridge" which is not what you have).

**Reply by pavkamlc** [reply]
Of course I tried it before, but without success. I think, problem must be on another place (bad flash layout from previous flashing?)
If I use upload, result is the some as ota.
Actualy my tries in console, there is better output:

14:59:27.837 CMD: otaurl
14:59:27.839 RSL: RESULT = {"OtaUrl":"[http://ota.tasmota.com/tasmota/tasmota-zbbridge.bin.gz"}](http://ota.tasmota.com/tasmota/tasmota-zbbridge.bin.gz%22%7D)
14:59:37.409 CMD: otaurl http://ota.tasmota.com/tasmota/tasmota-zbbridge.bin.gz
14:59:37.412 RSL: RESULT = {"OtaUrl":"[http://ota.tasmota.com/tasmota/tasmota-zbbridge.bin.gz"}](http://ota.tasmota.com/tasmota/tasmota-zbbridge.bin.gz%22%7D)
14:59:48.646 CMD: upload 1
15:00:05.510 RSL: UPGRADE = {"Upgrade":"Failed Update error: ERROR[8]: Flash config wrong real: 2097152 IDE: 4194304"}

**Comment by Jason2866** [comment, marked as answer]
Sonoff zbbridge was never flashed with Tuya convert since Sonoff has never had and has Tuya software on its devices. So you flashed somehow a wrong version. You have to serial flash the device, since you have a wrong image version build (4M) instead of the needed 2M build for the ZigBeeBridge.
